### PR TITLE
Using the proper system email keyboard on email input fields

### PR DESCRIPTION
**Pull request by macteo**
iOS and Android can propose to the user different keyboard types based on the requested input.

In the *Don't have SPID?* view, there is an email field that if selected presents the user a standard generic keyboard.
This commit addresses the issue:
* The presented keyboard is the email one.
* Auto capitalisation is turned off.
* Auto correction is turned off.
* To use those parameters, the `Form/Field` has been transformed into a `Form/TextInput`.

email-keyboard-ios
email-keyboard-android

### CODE REVIEW

**Comment by cloudify** [comment]
Hi @macteo, thanks for the PR! This is actually the first PR we get from an external contributor! congrats! 🏅 

I'd love to merge this, could you please fix the errors? thanks!

**Comment by fpersico** [comment]
I absolutely agree with you, we need to set a proper keyboard type, but i want to spot you some issues in your code and propose a different approach.

1. We are using redux-form so using the Field component is a requirement
2. We are using native-base components in the App and native-base Input component is just a "decorated" TextField
3. The react-native TextInput component doesen't accepts (so discards) some of the properties you are passing (for example 'validate')

So how to achieve the same result without breaking the above rules?

The redux-form Field component accepts a property to render the 'real' input field, in our case 'component={renderNativeBaseInput}'. If you check the 'renderNativeBaseInput' method inside `ts/components/forms/utils.tsx` you discover that it takes the Field component props and return an native-base Input component (wrapped inside Item). To pass additional parameters from the Field to the Input component you have to change the `NativeBaseInputProps` interface. So just add your desired properties to the interface:

```
export interface NativeBaseInputProps {
  name?: string;
  placeholder?: string;
  showError?: boolean;
  **keyboardType: KeyboardTypeOptions;**
}
```

than use the new property in renderNativeBaseInput:

```
export const renderNativeBaseInput: React.SFC<
  WrappedFieldProps & NativeBaseInputProps
> = ({
  meta: { touched, error, active },
  placeholder,
  showError,
  keyboardType
}) => (
  <View>
    <Item error={error && touched} active={active}>
      <Input
        placeholder={placeholder}
        keyboardType={keyboardType || "default"}
      />
    </Item>
    {showError && error && touched && <Text>{error}</Text>}
  </View>
);
```

Le me know what you think :)

File: ts/components/forms/utils.tsx [hunk 1]
```diff
@@ -44,6 +46,8 @@ export interface NativeBaseInputProps {
   name?: string;
   placeholder?: string;
   showError?: boolean;
+  keyboardType?: KeyboardTypeOptions;
+  autoCapitalize?: string;
```
Comment (fpersico) [inline ts/components/forms/utils.tsx]: autoCapitalize?: "none" | "sentences" | "words" | "characters";

File: ts/components/forms/SpidInformationForm.tsx [hunk 2]
```diff
@@ -25,6 +25,8 @@ class SpidInformationForm extends React.Component<InjectedFormProps, never> {
           placeholder={getCurrentFormFieldProperty("email")("placeholder")}
           validate={[validators.email]}
           showError={true}
+          keyboardType="email-address"
```
Comment (fpersico) [inline ts/components/forms/SpidInformationForm.tsx]: keyboardType={"email-address" as "email-address"}

File: ts/components/forms/SpidInformationForm.tsx [hunk 3]
```diff
@@ -20,15 +19,14 @@ class SpidInformationForm extends React.Component<InjectedFormProps, never> {
...
       <Form>
-        <TextInput
+        <Field
           name="email"
           component={renderNativeBaseInput}
           placeholder={getCurrentFormFieldProperty("email")("placeholder")}
           validate={[validators.email]}
           showError={true}
           keyboardType="email-address"
           autoCapitalize="none"
```
Comment (macteo) [inline ts/components/forms/SpidInformationForm.tsx]: autoCapitalize={"none" as "none"}

**Comment by macteo** [comment]
Arg, now the `yarn tsc:noemit` is successful, but `yarn lint` fails miserably due to a _unnecessary_ cast of `"email-address" as "email-address"` and `{"none" as "none"}`.

🐔 🥚

**Comment by macteo** [comment]
Disclaimer: the last commit is gross as I'm purposefully disabled lining on two lines.

**Comment by fpersico** [comment]
@macteo Please add

```
// TODO: Check wich type definition force us to disable tslint
// @https://www.pivotaltracker.com/story/show/157846397
```

before the line that disables tslint, so we can track this issue.

**Comment by fpersico** [comment]
@macteo Thanks! Your contribution in now in the master branch :tada:

**Comment by macteo** [comment]
🙏

**Pull request merged**